daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -1313,35 +1320,34 @@ private[lf] object Speedy {
     * directly into the environment.
     */
   private[speedy] final case class KPushTo(
-      machine: Machine,
+      savedBase: Int,
+      frame: Frame,
+      actuals: Actuals,
       to: util.ArrayList[SValue],
       next: SExpr,
   ) extends Kont
       with SomeArrayEquals {
-
-    private[this] val savedBase = machine.markBase()
-    private[this] val frame = machine.currentFrame
-    private[this] val actuals = machine.currentActuals
-
-    override def execute(v: SValue): Control = {
+    override def execute(machine: Machine, v: SValue): Control = {
       machine.restoreBase(savedBase);
       machine.restoreFrameAndActuals(frame, actuals)
       discard[Boolean](to.add(v))
       Control.Expression(next)
     }
   }
 
+  object KPushTo {
+    def apply(machine: Machine, to: util.ArrayList[SValue], next: SExpr): KPushTo =
+      KPushTo(machine.markBase(), machine.currentFrame, machine.currentActuals, to, next)
+  }
+
   private[speedy] final case class KFoldl(
-      machine: Machine,
+      frame: Frame,
+      actuals: Actuals,
       func: SValue,
       var list: FrontStack[SValue],
```

Comment:
Not for this PR, but I do feel rather uncomfortable with these `var`'s (especially as we're trying to allow the user of Speedy/Machine code layers to decide how threading should work here).

Reply:
This one is for @nickchapman-da.
 I think remember this was for performance reason.